File: csharp/sorting/InsertionSort.cs
using System;
using System.IO;
using System.Linq;

public class Program
{
	public static void Main(string[] args)
	{
		var path = args[0];
		
		int[] input;
		
		using (var reader = new StreamReader(path))
		{
			input = reader.ReadLine().Split(';').Select(x => Convert.ToInt32(x)).ToArray();	
		}
		
		var result = InsertionSort(input);

		foreach (var item in result)
			Console.Write(item + ";");
			
		Console.WriteLine();
	}
	
	private static int[] InsertionSort(int[] input)
	{
		for (var j = 1; j < input.Length; j++)
		{
			var key = input[j];
			
			var i = j - 1;
			while (i > -1 && input[i] > key)
			{
				input[i + 1] = input[i];
				i--;
			}
			
			input[i+1] = key;
		}
		
		return input;
	}
}

File: java/sorting/InsertionSort.java
import java.util.*;
import java.io.*;

public class InsertionSort {
	public static void main(String[] args) throws FileNotFoundException, IOException {
		String path = args[0];
		
		File file = new File(path);
		byte[] bytes = new byte[(int) file.length()];
		FileInputStream fis = new FileInputStream(file);
		fis.read(bytes);
		fis.close();
		
		String[] valueStr = new String(bytes).trim().split(";");
		int[] input = new int[valueStr.length];
		for (int i = 0; i < valueStr.length; i++)
			input[i] = Integer.parseInt(valueStr[i]);
		
		int[] result = InsertionSort(input);
		
		StringBuilder sb = new StringBuilder();		
		String prefix = "";
		
		for (int item : result) {
		  sb.append(prefix);
		  prefix = ";";
		  sb.append(item);
		}
		
		System.out.println(sb.toString());
	}
	
	private static int[] InsertionSort(int[] input) {
		for (int j = 1; j < input.length; j++) {
			int key = input[j];
			
			int i = j - 1;
			while (i > -1 && input[i] > key) {
				input[i + 1] = input[i];
				i--;
			}
			
			input[i+1] = key;
		}
		
		return input;
	}
}